lib/vmpooler/pool_manager.rb

```diff
@@ -603,6 +603,7 @@ def _check_pool(pool, provider)
         end
       rescue => err
         $logger.log('s', "[!] [#{pool['name']}] _check_pool failed with an error while inspecting inventory: #{err}")
+        return pool_check_response
```

Comment:
As I am swallowing the error in this case, the loop thread will not terminate so it will not need to be recreated